src/index/Index.cpp

```diff
@@ -438,7 +439,23 @@ void Index::exchangeMultiplicities(MetaData* m1, MetaData* m2) {
   }
 }
 
-// ____________________________________________________________________________
+// _____________________________________________________________________________
+void Index::addPatternsToExistingIndex() {
+  //  createPatternsImpl<MetaDataIterator<>>(
+  //      _onDiskBase + ".index.patterns", _vocab, _hasPredicate, _hasPattern,
+  //      _patterns, _fullHasPredicateMultiplicityEntities,
+  //      _fullHasPredicateMultiplicityPredicates, _fullHasPredicateSize,
+  //      _maxNumPatterns);
+
+  createPatternsImpl<MetaDataIterator<IndexMetaDataMmapView>,
+                     IndexMetaDataMmapView, ad_utility::File>(
+      _onDiskBase + ".index.patterns", _vocab, _hasPredicate, _hasPattern,
```

Comment:
I thik the `_vocab` is actually cleared in line 54 and is thus empty at this point?

Reply:
I merged your version of this pr. This now uses your code for determining the lower and upper bound.

Reply:
Awesome, I'm now trying to test your version on `wikidata-small` will report back when I know more.

Reply:
There seems to be a problem still, see on galera port 7042
```
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX wikibase: <http://wikiba.se/ontology#>
    
SELECT ?pred (COUNT(?pred) AS ?count) WHERE {
  ?human wdt:P31 wd:Q5 .
  ?human ql:has-predicate ?pred .
}
ORDER BY DESC(?count)
GROUP BY ?pred
```
still returns `@en@<stuff>`

Reply:
Weird. When testing locally I currently run into segfaults when running any query and trying to access the external vocabulary which apparently is never loaded. Where could I get the wikidata small dataset for testing?

Reply:
On galera you can look into `/local/data/wikidata-small` to see how I'm currently testing this with the still pre-alpha `gantry` tool. The wikidata-small set is in `/nfs/datasets/wikidata/small`.

From my current testing it looks like a problem with getting the correct string in `mergeVocabular()` I'm still debugging though.

Reply:
I accidentally checked the subject instead of the predicate agains the lang tag boundaries. The CreatPatternsMain executable also had a bug. Both of those should be fixed now.

Reply:
Thank you, I'm currently testing this on wikidata-small. I've also made a small change so we don't get a sign compare warning in the test.

Reply:
I get good results using the query you posted above after running the CreatePatternsMain executable on the index with the latest commit that fixed another small bug. I get a segfault when running on wikidata-truthy though.

Reply:
Apparently I just used an old docker image, the wikidata-truthy patterns rebuild is apparently going well now. The only thing I'm not sure about is how long it will take, given that it has to iterate over the entire MetaData and then load all entries of the SPO permuation